### PR TITLE
fix: skip tool_call deltas with function=None instead of crashing

### LINKED ISSUE #254: merge_deltas raises TypeError when a streamed tool_call delta has function=None
## Bug

When streaming tool-call deltas, a chunk can arrive where a `tool_call` delta has `function=None`. `merge_deltas` then raises `TypeError` (cannot un-pack/call `None`).

## Notes

- Documented as a known bug in PR #253 (test `test_tool_calls_function_is_none_raises`).
- The test docstring notes the fix belongs in a bug-fix PR; this issue is the tracking placeholder.
- Does not block PR #253, which only adds the test that reproduces the crash.

### PR DESCRIPTION
**Pull request by endolith**
## Background

In `run_tool_calling_llm`, the guard correctly skips converting tool_call deltas whose `function` is falsy — but the raw delta still reached `merge_deltas`, which calls `dict()` on the `tool_calls` list and raises `TypeError`.

## Problem

A streamed `tool_call` entry with `function=None` crashed the whole completion stream with `TypeError: cannot convert dictionary update sequence element`.

## Visible symptoms

`list(run_tool_calling_llm(...))` raised `TypeError` instead of yielding no chunks for the function-less delta.

## What this PR changes

- Drops the unprocessable `tool_calls` key so the chunk passes through empty, completing the guard's evident intent.
- Flips `test_tool_calls_function_is_none_raises` into `test_tool_calls_function_is_none_skipped` asserting `[]`.

## Tests

- `tests/core/llm/`: 164 passed.
- Full unit suite: 1037 passed, 32 skipped, 12 deselected (1 pre-existing xfail unrelated to this change).
- `ruff check` clean.

## Related work

Fixes #254.

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved streaming tool-call handling when function details are missing, preventing processing errors and allowing results to be consumed safely.
  - Added support for dictionary-formatted tool-call entries and valid details appearing after unusable entries.
  - Preserved nameless argument continuations when merging streamed tool calls.
  - Prevented malformed tool-call data from causing authentication checks to fail.

- **Tests**
  - Expanded edge-case coverage for incomplete, malformed, and dictionary-formatted tool-call data during streaming.

### CODE REVIEW

**Comment by endolith** [comment]
@coderabbit review

File: interpreter/core/llm/run_tool_calling_llm.py [hunk 1]
```diff
@@ -195,6 +195,16 @@ def run_tool_calling_llm(llm, request_params):
...
+            else:
+                # A tool_call entry without a function carries nothing the
+                # merge step can consume (merge_deltas cannot dict() the raw
+                # tool_calls list), so drop the key and let the chunk pass
+                # through empty rather than crashing.
+                delta = {
+                    key: value
+                    for key, value in delta.items()
+                    if key != "tool_calls"
+                }
```
[the remark this answers is not in the document]
Comment (endolith) [reply interpreter/core/llm/run_tool_calling_llm.py]: Fixed in 1014777d4e: restructured so `function_call_detected` is set only after a valid function is converted — a function-less delta now leaves the flag clear and yields nothing, so the auth guard can't false-trigger. Added `test_tool_calls_function_is_none_no_auth_error` with `INTERPRETER_REQUIRE_AUTHENTICATION=true` proving no raise and `[]` output.

— [opencode](https://github.com/anomalyco/opencode)

[the remark this answers is not in the document]
Comment (endolith) [reply interpreter/core/llm/run_tool_calling_llm.py]: Fixed in 1014777d4e: the handler now scans every entry by index and converts the first one carrying a usable function, instead of only inspecting entry 0. Entries are read shape-agnostically (`.function`/`.name`/`.arguments` attributes or dict keys) so this also survives the upcoming litellm upgrade. Added `test_tool_calls_later_valid_entry_converted` (None-function entry followed by a valid execute call yields the code chunk) and `test_tool_calls_dict_shaped_entries` (plain-dict entries convert the same way).

— [opencode](https://github.com/anomalyco/opencode)
